src/seedu/addressbook/data/person/Person.java

```diff
@@ -34,6 +36,35 @@ public Person(ReadOnlyPerson source) {
         this(source.getName(), source.getPhone(), source.getEmail(), source.getAddress(), source.getTags());
     }
 
+    /**
+     * Creates an exact copy of the person
+     * @return a copy of current person
+     */
+    public Person clonePerson() {
+        return new Person(getName(), getPhone(), getEmail(), getAddress(), getTags());
+    }
+
+    /**
+     * returns the person with smallest name in alphabetical order
+     * @param personList list of persons
+     * @return person with smallest name in alphabetical order
+     */
+    static Person pickSmallestPersonName(List<Person> personList) {
+        if (personList.isEmpty()) {
+            return null;
+        }
+        Iterator<Person> iterator = personList.iterator();
+        Person smallestPersonName = iterator.next();
+        Person nextPerson;
+        while (iterator.hasNext()) {
+            nextPerson = iterator.next();
+            if (nextPerson.isSmallerThanName(smallestPersonName)) {
+                smallestPersonName = nextPerson;
```

Comment:
seems a bit long, perhaps can increase more SLAP (Single-level of abstraction principle)?